openssl/src/crypto/hmac.rs

```diff
@@ -202,6 +190,73 @@ pub fn hmac(t: Type, key: &[u8], data: &[u8]) -> Result<Vec<u8>, ErrorStack> {
     h.finish()
 }
 
+#[cfg(ossl110)]
+#[allow(bad_style)]
+mod compat {
+    use ffi;
+
+    pub struct HMAC_CTX {
+        ctx: *mut ffi::HMAC_CTX,
+    }
+
+    impl HMAC_CTX {
+        pub fn new() -> HMAC_CTX {
+            unsafe {
+                let ctx = ffi::HMAC_CTX_new();
+                assert!(!ctx.is_null());
+                HMAC_CTX { ctx: ctx }
+            }
+        }
+
+        pub fn get(&self) -> *mut ffi::HMAC_CTX {
+            self.ctx
+        }
+    }
+
+    impl Drop for HMAC_CTX {
+        fn drop(&mut self) {
+            unsafe {
+                ffi::HMAC_CTX_free(self.ctx);
+            }
+        }
+    }
+}
+
+#[cfg(ossl10x)]
+#[allow(bad_style)]
+mod compat {
+    use std::mem;
+    use std::cell::UnsafeCell;
+
+    use ffi;
+
+    pub struct HMAC_CTX {
```

Comment:
Oh right, the struct is stored inline rather than being allocated by OpenSSL.

All of this stuff'll be deleted when we switch over to the PKey impl so it doesn't really matter.